pkg/cloudprovider/providers/gce/gce_loadbalancer.go

```diff
@@ -120,7 +127,8 @@ func (gce *GCECloud) EnsureLoadBalancer(clusterName string, apiService *v1.Servi
 		return nil, err
 	}
 
-	loadBalancerName := cloudprovider.GetLoadBalancerName(apiService)
+	loadBalancerName := gce.GetLoadBalancerName(apiService)
```

Comment:
Why? If a cloud provider calls ensureloadbalancer on a service, it will be updated with the new loadbalancer name, but if it does not call ensureloadbalancer(), it will use the DeprecatedLoadBalancerName(). Any other reason @justinsb that I am not folowing?